ramips: add support for TP-Link Archer C6 v3

amishv opened the pull request:
The patch adds support for the TP-Link Archer C6 v3 (FCC ID TE7A6V3)
The patch adds identification changes to the existing TP-Link Archer A6,
by Vinay Patil <post2vinay@gmail.com>, which has identical hardware.

Specification
-------------
MediaTek MT7621 SOC
RAM:         128MB DDR3
SPI Flash:   W25Q128 (16MB)
Ethernet:    MT7530 5x 1000Base-T
WiFi 5GHz:   Mediatek MT7613BE
WiFi 2.4GHz: Mediatek MT7603E
UART/Serial: 115200 8n1

Device Configuration & Serial Port Pins
---------------------------------------
ETH Ports:    LAN4 LAN3 LAN2 LAN1 WAN
             _______________________
             |                     |
Serial Pins: |   VCC GND TXD RXD   |
             |_____________________|

LEDs:         Power Wifi2G Wifi5G LAN WAN

Build Output
------------
The build will generate following set of files
[1] openwrt-ramips-mt7621-tplink_archer-c6-v3-initramfs-kernel.bin
[2] openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
[3] openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-sysupgrade.bin

How to Use - Flashing from TP-Link Web Interface
------------------------------------------------
* Go to "Advanced/System Tools/Firmware Update".
* Click "Browse" and upload the OpenWrt factory image: factory.bin[2]
* Click the "Upgrade" button, and select "Yes" when prompted.

TFTP Booting
------------
Setup a TFTP boot server with address 192.168.0.5.
While starting U-boot press '4' key to stop autoboot.
Copy the initramfs-kernel.bin[1] to TFTP server folder, rename as test.bin
From u-boot command prompt run tftpboot followed by bootm.

Recovery
--------
Archer A6 V3 has recovery page activated if SPI booting from flash fails.
Recovery page can be activated by powercycling the router four times
before the boot process is complete.
Note: TFTP boot can be activated only from u-boot serial console.
Device recovery address: 192.168.0.1

Signed-off-by: Amish Vishwakarma <vishwakarma.amish@gmail.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

blocktrron commented on target/linux/ramips/dts/mt7621_tplink_archer-c6-v3.dts:
```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
```
This DTS looks mostly identical to the A6-v3. The Ax/Cx series usually use the same board and only differ in the installed firmware.

Move identical nodes to a dtsi (mt7621_tplink_archer-x6-v3.dts) and include it in both devices.

amishv replied:
I was able to force A6 snapshot build once the C5 mods were working and found no difference. It seems only Signatures are different in the Flash. I was easily able to revert back to C6 by forcing the upgrade!

amishv replied:
Will make the change.
Thanks!

kpoman replied:
Greatly interested in testing. Have no UART or other unbricking tool so I wonder if this patch is safe to flash.

amishv replied:
> Greatly interested in testing. Have no UART or other unbricking tool so I wonder if this patch is safe to flash.

Archer C6 has a recovery mechanism, you need to power cycle it a few time in middle of booting to simulate boot failure and then you can upload the stock firmware using the interface on 192.168.0.1.

But, please, be aware that the router can brick even during the stock firmware upgrade!

blocktrron commented on target/linux/ramips/dts/mt7621_tplink_archer-c6-v3.dts:
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+
+#include "mt7621_tplink_archer-x6-v3.dtsi"
```
Only a single empty line before & after the include statement.

blocktrron commented on tools/firmware-utils/src/tplink-safeloader.c:
```diff
@@ -1061,6 +1061,47 @@ static struct device_info boards[] = {
+		.id     = "ARCHER-C6-V3",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
```
One tab too much.

amishv replied:
Corrected the tabs

blocktrron reviewed (changes requested):
Looks better now. Plese squash your added commit to your first one.

adschm commented:
Interesting. Are LAN and WAN MAC address really the same, which would be uncommon for TP-Link?

amishv commented:
> Interesting. Are LAN and WAN MAC address really the same, which would be uncommon for TP-Link?

The MAC addresses are stored in Flash. so each hardware device will have a unique MAC (similar to the serial no.) irrespective of the make and model

blocktrron commented:
@amishv Your commits are still not squashed.

amishv commented:
> @amishv Your commits are still not squashed.

I am facing multiple merge conflicts while trying to squash the commits. Kindly advise.

sairam commented:
https://patch-diff.githubusercontent.com/raw/openwrt/openwrt/pull/4222.patch

Use the patch from Github and force re-upload

adschm commented:
> > Interesting. Are LAN and WAN MAC address really the same, which would be uncommon for TP-Link?
> 
> The MAC addresses are stored in Flash. so each hardware device will have a unique MAC (similar to the serial no.) irrespective of the make and model

Of course, but is the MAC address of the LAN ports and the WAN port equal with stock firmware on the _same_ device?

amishv commented:
> > > Interesting. Are LAN and WAN MAC address really the same, which would be uncommon for TP-Link?
> > 
> > 
> > The MAC addresses are stored in Flash. so each hardware device will have a unique MAC (similar to the serial no.) irrespective of the make and model
> 
> Of course, but is the MAC address of the LAN ports and the WAN port equal with stock firmware on the _same_ device?

I check with another archer c1200 router, with stock firmware, using wireshark, it seems all the LAN and he WAN port share the same MAC.

amishv commented:
@adschm  Kindly remove the label target/ath79. This is MT7621 based device.

adschm commented:
> @adschm Kindly remove the label target/ath79. This is MT7621 based device.

Fact.

amishv commented:
> > @adschm Kindly remove the label target/ath79. This is MT7621 based device.
> 
> Fact.

compatible = "tplink,archer-c6-v3", "mediatek,mt7621-soc";
	model = "TP-Link Archer C6 v3";

amishv commented:
I am unable to get rid of _"hostapd: add default values for r0kh/r1kh "_  which somehow got merged with my code. 

I will be closing this PR and raise a new one.

adschm commented:
No, please don't open a new PR, but fix this one here.

amishv committed:
ramips: add support for TP-Link Archer C6 v3

The patch adds support for the TP-Link Archer C6 v3 (FCC ID TE7A6V3)
The patch adds identification changes to the existing TP-Link Archer A6,
by Vinay Patil <post2vinay@gmail.com>, which has identical hardware.

Specification
-------------
MediaTek MT7621 SOC
RAM:         128MB DDR3
SPI Flash:   W25Q128 (16MB)
Ethernet:    MT7530 5x 1000Base-T
WiFi 5GHz:   Mediatek MT7613BE
WiFi 2.4GHz: Mediatek MT7603E
UART/Serial: 115200 8n1

Device Configuration & Serial Port Pins
---------------------------------------
ETH Ports:    LAN4 LAN3 LAN2 LAN1 WAN
             _______________________
             |                     |
Serial Pins: |   VCC GND TXD RXD   |
             |_____________________|

LEDs:         Power Wifi2G Wifi5G LAN WAN

Build Output
------------
The build will generate following set of files
[1] openwrt-ramips-mt7621-tplink_archer-c6-v3-initramfs-kernel.bin
[2] openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
[3] openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-sysupgrade.bin

How to Use - Flashing from TP-Link Web Interface
------------------------------------------------
* Go to "Advanced/System Tools/Firmware Update".
* Click "Browse" and upload the OpenWrt factory image: factory.bin[2]
* Click the "Upgrade" button, and select "Yes" when prompted.

TFTP Booting
------------
Setup a TFTP boot server with address 192.168.0.5.
While starting U-boot press '4' key to stop autoboot.
Copy the initramfs-kernel.bin[1] to TFTP server folder, rename as test.bin
From u-boot command prompt run tftpboot followed by bootm.

Recovery
--------
Archer A6 V3 has recovery page activated if SPI booting from flash fails.
Recovery page can be activated by powercycling the router four times
before the boot process is complete.
Note: TFTP boot can be activated only from u-boot serial console.
Device recovery address: 192.168.0.1

Signed-off-by: Amish Vishwakarma <vishwakarma.amish@gmail.com>

amishv commented:
Fixed the git error and squashed old review commits!

kpoman commented:
Is there anything we could do to help here ? Any available binary around ? I dont see this one on snapshots repos !

amishv commented:
@kpoman the build will be available in snapshot once @blocktrron completes review and commits. 
If possible, please download the branch and build a snapshot.

kpoman commented:
> @kpoman the build will be available in snapshot once @blocktrron completes review and commits.
> If possible, please download the branch and build a snapshot.

Ok I got it compiled (downloading your branch). I got this:

```
kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ ll bin/targets/ramips/mt7621/
total 19800
drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:52 ./
drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:47 ../
-rw-r--r-- 1 kpoman kpoman     110 Jun  9 09:47 config.buildinfo
-rw-r--r-- 1 kpoman kpoman     263 Jun  9 09:47 feeds.buildinfo
-rw-r--r-- 1 kpoman kpoman 6594541 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-initramfs-kernel.bin
-rw-r--r-- 1 kpoman kpoman    2798 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3.manifest
-rw-r--r-- 1 kpoman kpoman 6826702 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
-rw-r--r-- 1 kpoman kpoman 6816833 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-sysupgrade.bin
drwxr-xr-x 2 kpoman kpoman    4096 Jun  9 09:52 packages/
-rw-r--r-- 1 kpoman kpoman     756 Jun  9 09:52 sha256sums
-rw-r--r-- 1 kpoman kpoman       8 Jun  9 09:47 version.buildinfo
kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ 

```
Do I just flash factory.bin via the original tplink firmware upgrade interface and wait ?

amishv commented:
> > @kpoman the build will be available in snapshot once @blocktrron completes review and commits.
> > If possible, please download the branch and build a snapshot.
> 
> Ok I got it compiled (downloading your branch). I got this:
> 
> ```
> kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ ll bin/targets/ramips/mt7621/
> total 19800
> drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:52 ./
> drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:47 ../
> -rw-r--r-- 1 kpoman kpoman     110 Jun  9 09:47 config.buildinfo
> -rw-r--r-- 1 kpoman kpoman     263 Jun  9 09:47 feeds.buildinfo
> -rw-r--r-- 1 kpoman kpoman 6594541 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-initramfs-kernel.bin
> -rw-r--r-- 1 kpoman kpoman    2798 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3.manifest
> -rw-r--r-- 1 kpoman kpoman 6826702 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
> -rw-r--r-- 1 kpoman kpoman 6816833 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-sysupgrade.bin
> drwxr-xr-x 2 kpoman kpoman    4096 Jun  9 09:52 packages/
> -rw-r--r-- 1 kpoman kpoman     756 Jun  9 09:52 sha256sums
> -rw-r--r-- 1 kpoman kpoman       8 Jun  9 09:47 version.buildinfo
> kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ 
> ```
> 
> Do I just flash factory.bin via the original tplink firmware upgrade interface and wait ?

Yes, use the original Firmware Upgrade page and upload Op*factory.bin. It should boot after upgrade.
If there is any issue set your LAN ip as 192.168.0.x and connect to 192.168.0.1 you will get the recovery page.

kpoman commented:
> > > @kpoman the build will be available in snapshot once @blocktrron completes review and commits.
> > > If possible, please download the branch and build a snapshot.
> > 
> > 
> > Ok I got it compiled (downloading your branch). I got this:
> > ```
> > kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ ll bin/targets/ramips/mt7621/
> > total 19800
> > drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:52 ./
> > drwxr-xr-x 3 kpoman kpoman    4096 Jun  9 09:47 ../
> > -rw-r--r-- 1 kpoman kpoman     110 Jun  9 09:47 config.buildinfo
> > -rw-r--r-- 1 kpoman kpoman     263 Jun  9 09:47 feeds.buildinfo
> > -rw-r--r-- 1 kpoman kpoman 6594541 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-initramfs-kernel.bin
> > -rw-r--r-- 1 kpoman kpoman    2798 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3.manifest
> > -rw-r--r-- 1 kpoman kpoman 6826702 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
> > -rw-r--r-- 1 kpoman kpoman 6816833 Jun  9 09:52 openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-sysupgrade.bin
> > drwxr-xr-x 2 kpoman kpoman    4096 Jun  9 09:52 packages/
> > -rw-r--r-- 1 kpoman kpoman     756 Jun  9 09:52 sha256sums
> > -rw-r--r-- 1 kpoman kpoman       8 Jun  9 09:47 version.buildinfo
> > kpoman@kpoman-ThinkPad-P50:~/Downloads/openwrt-tp-link-archer-c6-v3$ 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do I just flash factory.bin via the original tplink firmware upgrade interface and wait ?
> 
> Yes, use the original Firmware Upgrade page and upload Op*factory.bin. It should boot after upgrade.
> If there is any issue set your LAN ip as 192.168.0.x and connect to 192.168.0.1 you will get the recovery page.

Awesome, worked great, I am now installing luci and will test it ! Thanks !

kpoman commented:
Maybe unrelated, but apparently MT7613BE keeps crashing and only comes available if cold boot. Works for a couple minutes (or couple MB) then loses connectivity and is not detected anymore. I dont know if it is related to this build.

amishv commented:
> Maybe unrelated, but apparently MT7613BE keeps crashing and only comes available if cold boot. Works for a couple minutes (or couple MB) then loses connectivity and is not detected anymore. I dont know if it is related to this build.

I believe that is the issue with MT7613 drivers itself. I am facing the same issue on both Archer C6 and Linksys e5600. Will try RCA and file a defect in the MT7613 branch.

amishv commented:
> No, please don't open a new PR, but fix this one here.

@adschm  can you please help in completing the review and commit?

blocktrron commented:
Merged with changes to my staging tree. Thanks!

amishv commented on tools/firmware-utils/src/tplink-safeloader.c:
```diff
@@ -1061,6 +1061,47 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system",
 	},
+	/** Firmware layout for the Archer C6 v3 */
+    {
+		.id     = "ARCHER-C6-V3",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+		        "{product_name:Archer C6,product_ver:3.20,special_id:55530000}"
+            		"{product_name:Archer C6,product_ver:3.20,special_id:45550000}"
+            		"{product_name:Archer C6,product_ver:3.20,special_id:52550000}"
+            		"{product_name:Archer C6,product_ver:3.20,special_id:4A500000}"
+            		"{product_name:Archer C6,product_ver:3.20,special_id:4B520000}",
+		.part_trail = 0x00,
+		.soft_ver = "soft_ver:1.0.9\n",
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x40000},
+			{"firmware", 0x40000, 0xf60000},
+			{"default-mac", 0xfa0000, 0x00200},
+			{"pin", 0xfa0200, 0x00100},
+			{"device-id", 0xfa0300, 0x00100},
+			{"product-info", 0xfa0400, 0x0fc00},
+			{"default-config", 0xfb0000, 0x08000},
+			{"ap-def-config", 0xfb8000, 0x08000},
+			{"user-config", 0xfc0000, 0x0a000},
+			{"ag-config", 0xfca000, 0x04000},
```
thanks @blocktrron!

amishv commented:
> Merged with changes to my staging tree. Thanks!

@blocktrron  when are we planning to commit this to the master? I aam hoping to get it into 21.02 after getting the snapshot build tested.

adschm commented on target/linux/ramips/dts/mt7621_tplink_archer-x6-v3.dtsi:
```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
@blocktrron Looks like this one has been relicensed compared to the source of the copy-paste ...?

amishv replied:
> @blocktrron Looks like this one has been relicensed compared to the source of the copy-paste ...?

@adschm, thanks for pointing out,  this was perhaps introduced when I had issues with squashing multiple comment (due to a deleted commit on which my tree was branched) . The license should not be changed.

rafizulhasan commented:
sir any new build come? i want to test ur build, plz give me a try, provide me ur telegram id

kpoman commented:
I just tried to build with image builder, with this command:
`make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
However, bizarely, I got an adslr-g7 image ! See below:
```
kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
kpoman@kpoman-ThinkPad-X230
```

Weird !
kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
Building images for ramips - ADSLR G7

kpoman commented:
> I just tried to build with image builder, with this command:
> `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> However, bizarely, I got an adslr-g7 image ! See below:
> 
> ```
> kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> kpoman@kpoman-ThinkPad-X230
> ```
> 
> Weird !
> kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> Building images for ramips - ADSLR G7

Sorry, I am pretty dumb, put PROFILE now and it works.

rafizulhasan commented:
> > I just tried to build with image builder, with this command:
> > `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> > However, bizarely, I got an adslr-g7 image ! See below:
> > ```
> > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> > openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> > openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> > kpoman@kpoman-ThinkPad-X230
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Weird !
> > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> > Building images for ramips - ADSLR G7
> 
> Sorry, I am pretty dumb, put PROFILE now and it works.

sir is ur build is stable? if then plz give me link, i want to flash

kpoman commented:
> > > I just tried to build with image builder, with this command:
> > > `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> > > However, bizarely, I got an adslr-g7 image ! See below:
> > > ```
> > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> > > openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> > > openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> > > kpoman@kpoman-ThinkPad-X230
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Weird !
> > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> > > Building images for ramips - ADSLR G7
> > 
> > 
> > Sorry, I am pretty dumb, put PROFILE now and it works.
> 
> sir is ur build is stable? if then plz give me link, i want to flash

You should be able to download here:
https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin

rafizulhasan commented:
> > > > I just tried to build with image builder, with this command:
> > > > `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> > > > However, bizarely, I got an adslr-g7 image ! See below:
> > > > ```
> > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> > > > openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> > > > openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> > > > kpoman@kpoman-ThinkPad-X230
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Weird !
> > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> > > > Building images for ramips - ADSLR G7
> > > 
> > > 
> > > Sorry, I am pretty dumb, put PROFILE now and it works.
> > 
> > 
> > sir is ur build is stable? if then plz give me link, i want to flash
> 
> You should be able to download here:
> https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin

sir any bug or it will be usable ? plz

rafizulhasan commented:
> 
> 
> > > > I just tried to build with image builder, with this command:
> > > > `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> > > > However, bizarely, I got an adslr-g7 image ! See below:
> > > > ```
> > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> > > > openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> > > > openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> > > > kpoman@kpoman-ThinkPad-X230
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Weird !
> > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> > > > Building images for ramips - ADSLR G7
> > > 
> > > 
> > > Sorry, I am pretty dumb, put PROFILE now and it works.
> > 
> > 
> > sir is ur build is stable? if then plz give me link, i want to flash
> 
> You should be able to download here:
> https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin

![Screenshot_1](https://user-images.githubusercontent.com/64944207/126074740-eeff1841-9722-466f-a2fa-7959bcee1ec3.png)
sir can i try this? plz replay

kpoman commented:
> > > > > I just tried to build with image builder, with this command:
> > > > > `make image profile=tplink_archer-c6-v3 PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo -wpad-basic-wolfssl"`
> > > > > However, bizarely, I got an adslr-g7 image ! See below:
> > > > > ```
> > > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ ll bin/targets/ramips/mt7621/
> > > > > openwrt-ramips-mt7621-adslr_g7.manifest                 profiles.json                                           
> > > > > openwrt-ramips-mt7621-adslr_g7-squashfs-sysupgrade.bin  sha256sums                                              
> > > > > kpoman@kpoman-ThinkPad-X230
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > Weird !
> > > > > kpoman@kpoman-ThinkPad-X230 ~/openwrt/openwrt-imagebuilder-ramips-mt7621.Linux-x86_64 $ make image profile="tplink_archer-c6-v3" PACKAGES="luci wpad-mesh-wolfssl zram-swap collectd collectd-mod-network collectd-mod-cpu collectd-mod-iwinfo"
> > > > > Building images for ramips - ADSLR G7
> > > > 
> > > > 
> > > > Sorry, I am pretty dumb, put PROFILE now and it works.
> > > 
> > > 
> > > sir is ur build is stable? if then plz give me link, i want to flash
> > 
> > 
> > You should be able to download here:
> > https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/openwrt-ramips-mt7621-tplink_archer-c6-v3-squashfs-factory.bin
> 
> ![Screenshot_1](https://user-images.githubusercontent.com/64944207/126074740-eeff1841-9722-466f-a2fa-7959bcee1ec3.png)
> sir can i try this? plz replay

You can try it yes. The only problem for me right now is on 5GHz with multiple stations.
When you install this, you'll have to connect to the router via ssh, because snapshots dont come with luci admin interface. No big deal, just opkg update && opkg install luci !

galacticwarrior448 commented:
I tested out your build but the 5ghz speed is horrible can anyone help. I have a 300mbps connection. And I get 10 in 5Ghz, 2.4ghz works fine.

SamKur commented:
Archer C6 and a6 (talking about v3.2) have same hardware, same official firmware, same gpl code (visit a6 download page and see the filename too). 

So, it's safe to install archer a6 openwrt firmware to archer C6 , i guess. I'm trying to install via official firmware upgrade page, but unable to upgrade. Also, tried with TFTP but, unable to do , it's not reading for any file in reset mode. 

Anyone please help me upgrading.
